include/NovelRT/Maths/GeoVector2F.h

```diff
@@ -21,157 +24,789 @@ namespace NovelRT::Maths
+         */
         inline bool operator>(GeoVector2F other) const noexcept
         {
             return glm::any(glm::greaterThan(*reinterpret_cast<const glm::vec2*>(this),
                                              *reinterpret_cast<const glm::vec2*>(&other)));
         }
 
+        /**
+         * @brief Evaluates this GeoMatrix4x4F with another GeoMatrix4x4F to determine if any of it's components is
+         * greater than or equal to the other's components.
+         *
+         * @param other The GeoMatrix4x4F who's components are evaluated against this GeoMatrix4x4F's components.
+         * @return true if any of the components is greater than or equal to the other GeoMatrix4x4F's components,
+         * otherwise false.
+         */
         inline bool operator>=(GeoVector2F other) const noexcept
         {
             return glm::any(glm::greaterThanEqual(*reinterpret_cast<const glm::vec2*>(this),
                                                   *reinterpret_cast<const glm::vec2*>(&other)));
         }
 
+        /**
+         * @brief Constructs a GeoVector2F containing the entrywise sum of this and another GeoVector2F.
+         *
+         * @details
+         * An example of an entrywise addition using 2 dimensional vectors: \f[
+         *      \begin{pmatrix}
+         *      2\\
+         *      3
+         *      \end{pmatrix}
+         *      +
+         *      \begin{pmatrix}
+         *      4\\
+         *      5
+         *      \end{pmatrix}
+         *      =
+         *      \begin{pmatrix}
+         *      2 + 4\\
+         *      3 + 5
+         *      \end{pmatrix}
+         *      \rightarrow
+         *      \begin{pmatrix}
+         *      6\\
+         *      8
+         *      \end{pmatrix}
+         * \f]
+         *
+         * @param other The other vector who's components get added together with this vector.
+         * @return A GeoVecor2F who's components consist of the entrywise sum of this and another GeoVector2F.
+         */
         inline GeoVector2F operator+(GeoVector2F other) const noexcept
         {
             return GeoVector2F(*reinterpret_cast<const glm::vec2*>(this) + *reinterpret_cast<const glm::vec2*>(&other));
         }
 
+        /**
+         * @brief Constructs a GeoVector2F containing the entrywise difference of this and another GeoVector2F.
+         *
+         * @details
+         * An example of an entrywise subtraction using 2 dimensional vectors: \f[
+         *      \begin{pmatrix}
+         *      2\\
+         *      3
+         *      \end{pmatrix}
+         *      -
+         *      \begin{pmatrix}
+         *      4\\
+         *      5
+         *      \end{pmatrix}
+         *      =
+         *      \begin{pmatrix}
+         *      2 - 4\\
+         *      3 - 7
+         *      \end{pmatrix}
+         *      \rightarrow
+         *      \begin{pmatrix}
+         *      -2\\
+         *      -4
+         *      \end{pmatrix}
+         * \f]
```

Comment:
Can you explain the use of matrix stuff inside of the vector documentation?

Reply:
That has to do with how LaTeX constructs formulas, here is an example of how it would look like. 
![image](https://user-images.githubusercontent.com/22686676/182368443-8e6b8cad-8d4f-4ec7-8f8e-d2d32e3d7a55.png)

TL;DR `pmatrix` is for putting stuff in `()`, `bmatrix` is for putting stuff in `[]`, i use `()` when talking about vectors and `[]` when i am talking about matrices.